runtime/src/iree/io/formats/safetensors/safetensors_format.h

```diff
@@ -0,0 +1,41 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_IO_FORMATS_SAFETENSORS_SAFETENSORS_FORMAT_H_
+#define IREE_IO_FORMATS_SAFETENSORS_SAFETENSORS_FORMAT_H_
+
+#include "iree/base/api.h"
+#include "iree/io/file_handle.h"
+#include "iree/io/parameter_index.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+// Parses a .safetensors file and merges its contained resources into |index|.
+//
+// Documentation: https://github.com/huggingface/safetensors
+// This is a very basic archive file with some issues (no alignment, etc) but
+// at least doesn't require Python pickle decoding (just JSON). The major reason
+// to use this is if sourcing from a Hugging Face model that has its weights
+// already in the safetensors format.
+//
+// WARNING: this implementation has not been thoroughly tested or verified as
+// safe or correct. Use with caution only on trusted inputs. Tip: don't embed
```

Comment:
/me adds to list of tests to add fuzzers for at some point

(edit: read the _format.c... oh... hmm 🤔)